module/modmanager/manager.go

```diff
@@ -1074,6 +1075,49 @@ func (m *module) checkReady(ctx context.Context, parentAddr string, logger loggi
 	}
 }
 
+// FirstRun is runs a module-specific setup script.
+// TODO: return exit code?
+func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+	// We evaluate the Module's ExePath absolutely in the viam-server process so that
+	// setting the CWD does not cause issues with relative process names
+	firstRunPath, err := conf.EvaluateFirstRunPath(packages.LocalPackagesDir(mgr.packagesDir))
+	if err != nil {
+		return err
+	}
+
+	// TODO: this is normally set on a module but it seems like we can safely get it on demand.
+	var dataDir string
+	if mgr.moduleDataParentDir != "" {
+		var err error
+		// TODO: why isn't conf.Name being sanitized like PackageConfig.SanitizedName?
+		dataDir, err = rutils.SafeJoinDir(mgr.moduleDataParentDir, conf.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	moduleEnvironment := getFullEnvironment(conf, dataDir, mgr.viamHomeDir)
+
+	// TODO: Revert this nolint declaration. I'm temporarily ignoring this lint for
+	// testing purposes - we don't want to disable this lint since it opens an obvious
+	// injection attack vector.
+	//nolint:gosec
+	cmd := exec.CommandContext(ctx, firstRunPath)
```

Comment:
the lint is legit and though we discussed it IRL already, I just want to point it out again in case reviewers have second thoughts about this. we could consider adding _some_ sanitation here.